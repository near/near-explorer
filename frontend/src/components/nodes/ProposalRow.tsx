import React from "react";
import { Row, Col, OverlayTrigger, Tooltip } from "react-bootstrap";

import * as N from "../../libraries/explorer-wamp/nodes";
import Balance from "../utils/Balance";

interface Props {
  node: N.Proposal;
}

export default class extends React.PureComponent<Props> {
  render() {
    const { node } = this.props;
    return (
      <Row className="node-row mx-0">
        <Col md="auto" xs="1" className="pr-0">
          <OverlayTrigger
            placement={"right"}
            overlay={
              <Tooltip id="nodes">node staked to be new validating one</Tooltip>
            }
          >
            <p>
              <img
                src={"/static/images/icon-m-node-proposal.svg"}
                style={{ width: "15px", verticalAlign: "text-top" }}
              />
            </p>
          </OverlayTrigger>
        </Col>
        <Col md="7" xs="7">
          <Row>
            <Col className="node-row-title">
              <p>
                @{node.account_id}
                {"  "}
                <span>
                  Staking {node.stake ? <Balance amount={node.stake} /> : "-"}
                </span>
              </p>
            </Col>
          </Row>
        </Col>
        <Col md="3" xs="3" className="ml-auto text-right">
          <Row>
            <p>
              <Col className="node-row-txid" title={node.public_key}>
                {node.public_key.substring(8, 20)}...
              </Col>
            </p>
          </Row>
        </Col>
        <style jsx global>{`
          .node-row {
            padding-top: 10px;
            border-top: solid 2px #f8f8f8;
          }

          .node-row:hover {
            background: rgba(0, 0, 0, 0.1);
          }

          .node-row-bottom {
            border-bottom: solid 2px #f8f8f8;
          }

          .node-row-title {
            font-family: BentonSans;
            font-size: 14px;
            font-weight: 500;
            line-height: 1.29;
            color: #24272a;
          }

          .node-row-txid {
            font-family: BentonSans;
            font-size: 14px;
            font-weight: 500;
            line-height: 1.29;
            color: #4a4f54;
          }
        `}</style>
      </Row>
    );
  }
}
