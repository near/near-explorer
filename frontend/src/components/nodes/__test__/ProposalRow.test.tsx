import renderer from "react-test-renderer";

import ProposalRow from "../ProposalRow";

describe("<ProposalRow />", () => {
  it("renders", () => {
    expect(
      renderer.create(
        <ProposalRow
          node={{
            account_id: "dokiacapital.pool.6fb1358",

            public_key: "ed25519:935JMz1vLcJxFApG3TY4MA4RHhvResvoGwCrQoJxHPn9",

            stake: "91037770393145811562101780866",
          }}
        />
      )
    ).toMatchSnapshot();
  });
});
