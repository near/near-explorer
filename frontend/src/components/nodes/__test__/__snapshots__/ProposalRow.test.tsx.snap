// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProposalRow /> renders 1`] = `
<div
  className="node-row mx-0 row"
>
  <div
    className="pr-0 col-md-auto col-1"
  >
    <p
      className="jsx-2919956104"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <img
        className="jsx-2919956104"
        src="/static/images/icon-m-node-proposal.svg"
        style={
          Object {
            "verticalAlign": "text-top",
            "width": "15px",
          }
        }
      />
    </p>
  </div>
  <div
    className="col-md-7 col-7"
  >
    <div
      className="row"
    >
      <div
        className="node-row-title col"
      >
        <p
          className="jsx-2919956104"
        >
          @
          dokiacapital.pool.6fb1358
            
          <span
            className="jsx-2919956104"
          >
            Staking 
            <span
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              91,037.77039
               â“ƒ
            </span>
          </span>
        </p>
      </div>
    </div>
  </div>
  <div
    className="ml-auto text-right col-md-3 col-3"
  >
    <div
      className="row"
    >
      <p
        className="jsx-2919956104"
      >
        <div
          className="node-row-txid col"
          title="ed25519:935JMz1vLcJxFApG3TY4MA4RHhvResvoGwCrQoJxHPn9"
        >
          935JMz1vLcJx
          ...
        </div>
      </p>
    </div>
  </div>
</div>
`;
